src/daemon/daemon_config.cpp

```diff
@@ -162,7 +163,7 @@ std::unique_ptr<const mp::DaemonConfig> mp::DaemonConfigBuilder::build()
                                                               server_name_from(server_address));
     if (client_cert_store == nullptr)
         client_cert_store =
-            std::make_unique<mp::ClientCertStore>(mp::utils::make_dir(data_directory, "registered-certs"));
+            std::make_unique<mp::ClientCertStore>(mp::utils::make_dir(data_directory, mp::registered_certs_dir));
```

Comment:
Hmm, I think the `ClientCertStore` should be the one responsible for determining the permissions and setting them correctly since it is the entity entirely responsible for it.

Reply:
Sure, and that's where this `make_dir` should go, along with the new `permissions` argument :)

Reply:
Ok, this has been changed.